pydm/widgets/base.py

```diff
@@ -698,8 +730,34 @@ def channel(self, value):
             Channel address
         """
         if self._channel != value:
+            # Remove old connections
+            for channel in self._channels:
+                if channel.address == self._channel:
+                    channel.disconnect()
+                    self._channels.remove(channel)
```

Comment:
Is this modifying the list being iterated over?

Reply:
It is. This will be fixed...

Reply:
Still has not been addressed

Reply:
@hhslepicka Want me to change this or you?